test/common/config/grpc_subscription_test_harness.h

```diff
@@ -44,7 +44,7 @@ class GrpcSubscriptionTestHarness : public SubscriptionTestHarness {
             "envoy.api.v2.EndpointDiscoveryService.StreamEndpoints")),
         async_client_(new NiceMock<Grpc::MockAsyncClient>()) {
     node_.set_id("fo0");
-    EXPECT_CALL(local_info_, node()).WillOnce(testing::ReturnRef(node_));
+    EXPECT_CALL(local_info_, node()).WillRepeatedly(testing::ReturnRef(node_));
```

Comment:
I think we should add some tests to `ads_integration_test` to capture this regression end-to-end, for both SotW and delta. This will provide confidence that we don't hit this issue again.

Reply:
Do you know if there's already a way to reset the stream during an integration test? Also sorry, this might take me a while because I've never written an integration test

Reply:
Sure, you can just close the fake upstream corresponding to the ADS server. See `xds_upstream_`, which you can access from `ads_integration_test.cc`. You can probably use something like `encodeResetStream()` on the ADS stream.

Reply:
I think I got something working but should probably merge #14270 first